code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -127,10 +129,10 @@ const styles = {
  */
 const ProgressDot = React.createClass({
   propTypes: {
-    levels: levelProgressShape,
+    level: levelProgressShape.isRequired,
     currentLevelId: React.PropTypes.string,
     largeDots: React.PropTypes.bool,
-    saveAnswersFirst: React.PropTypes.bool.isRequired
+    saveAnswersBeforeNavigation: React.PropTypes.bool.isRequired
```

Comment:
I'd add a test for the `ProgressDot` component.  For connected components, I've been exporting the unconnected version too and testing that.

```javascript
// progress_dot.jsx
export const ProgressDot = React.createClass({ /* ... */ });
export default connect(/* ... */)(ProgressDot);

// progress_dot_test.jsx
import { ProgressDot } from '...progress_dot.jsx'; // Get the unconnected version
// test in isolation.
```

[AppLabTooltipOverlayTest](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/applab/AppLabTooltipOverlayTest.js) more or less follows this pattern.